src/components/calculator.js

```diff
@@ -2,41 +2,128 @@
 import React from 'react';
 import Button from './button';
 import './calculator.css';
+import calculate from '../logic/calculate';
 
 export default class Calculator extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      obj: { total: null, next: null, operation: null },
+    };
+    this.handleClick = this.handleClick.bind(this);
+  }
+
+  handleClick(buttonName) {
+    const { obj } = this.state;
+    const newObj = {
+      ...obj,
+      ...calculate(obj, buttonName),
+    };
+    this.setState({ obj: newObj });
+  }
+
   render() {
+    const { obj } = this.state;
+    const res = obj.next ? obj.next : obj.total;
     return (
       <div className="calculator-wrap">
-        <input
-          type="text"
-          name="result"
-          id="res"
-          value={0}
-          readOnly
-        />
-        <Button position="main" text="AC" />
-        <Button position="main" text="+/-" />
-        <Button position="main" text="%" />
-        <Button position="aside" text="+" />
-        <Button position="main" text="7" />
-        <Button position="main" text="8" />
-        <Button position="main" text="9" />
-        <Button position="aside" text="&times;" />
-        <Button position="main" text="4" />
-        <Button position="main" text="5" />
-        <Button position="main" text="6" />
-        <Button position="aside" text="-" />
-        <Button position="main" text="1" />
-        <Button position="main" text="2" />
-        <Button position="main" text="3" />
-        <Button position="aside" text="+" />
+        <div id="res">{Number(res)}</div>
+        <Button
+          position="main"
+          text="AC"
+          onButtonClick={this.handleClick}
+        />
+        <Button
+          position="main"
+          text="+/-"
+          onButtonClick={this.handleClick}
+        />
+        <Button
+          position="main"
+          text="%"
+          onButtonClick={this.handleClick}
+        />
+        <Button
+          position="aside"
+          text="÷"
+          onButtonClick={this.handleClick}
```

Comment:
- [ ] When I try to perform **modulus** and **divide** calculations using **`0`** denominator, I get a **`NaN`** error (see image). However, according to the `operate.js` file, this should return `Can't find modulo as can't divide by 0.` and `Can't divide by 0.` messages instead. Kindly fix this.

  ![boursen-divide](https://user-images.githubusercontent.com/36542986/158596833-fdeb2df0-63d0-4ca8-ad4e-6fe5173ae0e0.jpg)

Reply:
I think this should be more of a suggestion/recommendation and not a requirement.
There is no requirement saying that we should return a custom-made error message instead of **NAN**.
Please refer to the [requirement list](https://github.com/microverseinc/curriculum-react-redux/blob/main/math-magicians/project_events.md)

I will try to implement it nonetheless because it's a good **suggestion**